observer/probers/crl/crl_conf.go

```diff
@@ -0,0 +1,132 @@
+package probers
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/letsencrypt/boulder/observer/probers"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	nextUpdateName = "obs_crl_next_update"
+	thisUpdateName = "obs_crl_this_update"
+	certCountName  = "obs_crl_revoked_cert_count"
+)
+
+// CRLConf is exported to receive YAML configuration
+type CRLConf struct {
+	URL string `yaml:"url"`
+}
+
+// Kind returns a name that uniquely identifies the `Kind` of `Configurer`.
+func (c CRLConf) Kind() string {
+	return "CRL"
+}
+
+// UnmarshalSettings constructs a CRLConf object from YAML as bytes.
+func (c CRLConf) UnmarshalSettings(settings []byte) (probers.Configurer, error) {
+	var conf CRLConf
+	err := yaml.Unmarshal(settings, &conf)
+	if err != nil {
+		return nil, err
+	}
+	return conf, nil
+}
+
+func (c CRLConf) validateURL() error {
+	url, err := url.Parse(c.URL)
+	if err != nil {
+		return fmt.Errorf(
+			"invalid 'url', got: %q, expected a valid url", c.URL)
+	}
+	if url.Scheme == "" {
+		return fmt.Errorf(
+			"invalid 'url', got: %q, missing scheme", c.URL)
+	}
+	return nil
+}
+
+// MakeProber constructs a `CRLProbe` object from the contents of the
+// bound `CRLConf` object. If the `CRLConf` cannot be validated, an
+// error appropriate for end-user consumption is returned instead.
+func (c CRLConf) MakeProber(collectors map[string]prometheus.Collector) (probers.Prober, error) { // validate `url` err := c.validateURL()
+	// validate `url`
+	err := c.validateURL()
+	if err != nil {
+		return nil, err
+	}
+	probe := CRLProbe{c.URL, nil, nil, nil}
+
+	// validate the prometheus collectors that were passed in
+	for name, coll := range collectors {
+		var ok bool
```

Comment:
Declare this once, outside of the loop on line `62`.
```suggestion
		var ok bool
```

Reply:
done